pym/portage/package/ebuild/_parallel_manifest/ManifestScheduler.py

```diff
@@ -91,3 +81,55 @@ def _task_exit(self, task):
 					noiselevel=-1)
 
 		AsyncScheduler._task_exit(self, task)
+
+
+def _future_fetchlist(loop, portdb, repo_config, cp, cpv_list):
+	"""
+	Asynchronous form of FetchlistDict.
+
+	@param loop: event loop
+	@type loop: EventLoop
+	@param portdb: portdbapi instance
+	@type portdb: portdbapi
+	@param repo_config: repository configuration for a Manifest
+	@type repo_config: RepoConfig
+	@param cp: cp for a Manifest
+	@type cp: str
+	@param cpv_list: list of ebuild cpv values for a Manifest
+	@type cpv_list: list
+	@return: a Future resulting in a Mapping compatible with FetchlistDict
+	@rtype: asyncio.Future (or compatible)
+	"""
+	loop = getattr(loop, '_asyncio_wrapper', loop)
+	result = loop.create_future()
+	futures = {}
+	for cpv in cpv_list:
+		futures[cpv] = portdb.async_fetch_map(
+			cpv, mytree=repo_config.location, loop=loop)
```

Comment:
We need so use something like `async_iter_completed(max_jobs=1)` to limit the number of concurrent async_fetch_map calls here, otherwise the only limit is the number of ebuilds in the directory.

Reply:
Fixed to use `async_iter_completed(max_jobs=1)`.